Application/Features/Queries/GetReservationByMovieId/GetReservationByMovieIdQueryHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Application.Contracts;
+using Application.Exceptions;
+using Application.Interfaces;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Features.Queries.GetReservationByMovieId;
+
+public sealed class GetReservationByMovieIdQueryHandler : IRequestHandler<GetReservationByMovieIdQuery, ReturnReservationDto>
+{
+    private readonly IDataContext _context;
+
+    public GetReservationByMovieIdQueryHandler(IDataContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<ReturnReservationDto> Handle(GetReservationByMovieIdQuery request, CancellationToken cancellationToken)
+    {
+        var reservation =
+            await _context.Reservations
+                .SingleOrDefaultAsync(x => x.MovieId == request.MovieId, new CancellationToken());
```

Comment:
Imamo postojeći cancellation token kao parametar - u suštini gotovo nikad ne bi trebalo ovako da pravimo novi samo da bi poslali ako je opcioni parametar (kao što jeste za SingleOrDefaultAsync)